### PR TITLE
kustomize: add deprecation warning to docs

### PR DESCRIPTION
**Pull request by natasha41575**
Should be merged after release containing the deprecation warnings

/hold
/cc @KnVerey

### CODE REVIEW

**Comment by natasha41575** [comment]
I need to update this PR to reflect all the deprecated fields, and change the version number.

**Comment by holdeneoneal** [comment]
@natasha41575 can you comment on when these features will be removed from kustomize? Is there an issue or milestone for their planned removal? Looking to understand what our runway is to remove these from our code base. Thank you!

**Comment by natasha41575** [comment]
@holdeneoneal Thank you for the very valid question! In response to your question, I added the answer to the docs in this PR so that it is really clear what the timeline is, which applies to all deprecated fields:

> Though this field will never be removed from the current Kustomization v1beta1 API, the field will be removed when we
define the API for Kustomization v1, at which time we will announce that Kustomization v1beta1 is deprecated. There will be at least two releases between deprecation and removal
of Kustomization v1beta1.

There is no strict timeline for this, but realistically the removal of these fields will likely take at least a year or two. It is only when we have deprecated Kustomization v1beta1 entirely that we will be in a position to drop support for deprecated fields completely, so you should keep an eye out for that announcement.

**Comment by natasha41575** [comment]
@KnVerey This is ready for review!

**Comment by natasha41575** [comment]
/hold

So that this isn't merged until release

**Comment by holdeneoneal** [comment]
> @holdeneoneal Thank you for the very valid question! In response to your question, I added the answer to the docs in this PR so that it is really clear what the timeline is, which applies to all deprecated fields:
> 
> > Though this field will never be removed from the current Kustomization v1beta1 API, the field will not be included
> > in the API for Kustomization v1. When Kustomization v1 is available, we will announce the deprecation of Kustomization v1beta1. There will be at least two releases between deprecation and removal of Kustomization v1beta1.
> 
> There is no strict timeline for this, but realistically the removal of these fields will likely take at least a year or two. It is only when we have deprecated Kustomization v1beta1 entirely that we will be in a position to drop support for deprecated fields completely, so you should keep an eye out for that announcement.

Thank you, Natasha! I have a follow up question. When you say "at least two releases", are those minor version releases of kustomize or major version releases?

**Comment by natasha41575** [comment]
> Thank you, Natasha! I have a follow up question. When you say "at least two releases", are those minor version releases of kustomize or major version releases?

Any two releases, I think this includes patch/minor releases. Kustomize is part of kubectl, so we generally have to follow their deprecation policy: https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecating-a-flag-or-cli

**Comment by koba1t** [comment]
Thanks, @natasha41575!
I think this announce is very useful!

So, I think the [legacy patches](https://github.com/kubernetes-sigs/kustomize/pull/4911) syntax is removed next release.
Could you consider announcing this syntax change?

File: site/content/en/references/kustomize/kustomization/bases/_index.md [hunk 1]
```diff
@@ -8,7 +8,11 @@ description: >
 ---
 
 {{% pageinfo color="warning" %}}
-The `bases` field was deprecated in v2.1.0
+The `bases` field was deprecated in v2.1.0. Though this field will never be
+removed from the current Kustomization v1beta1 API, the field will not be included
+in the API for Kustomization v1. When Kustomization v1 is available, we will announce
+the deprecation of Kustomization v1beta1. There will be at least two releases between
+deprecation and removal of Kustomization v1beta1.
```
Comment (KnVerey) [inline site/content/en/references/kustomize/kustomization/bases/_index.md:15]: nit: maybe also clarify that the v1beta1 removal itself will happen as part of a future major version bump? 

Here's what we went with in the release notes: https://github.com/kubernetes-sigs/kustomize/releases/tag/kustomize%2Fv5.0.0

**Comment by KnVerey** [comment]
> So, I think the https://github.com/kubernetes-sigs/kustomize/pull/4911 syntax is removed next release.
Could you consider announcing this syntax change?

Great point! That syntax is super old and currently nowhere in the docs, but we added its removal to the [release notes](https://github.com/kubernetes-sigs/kustomize/releases/tag/kustomize%2Fv5.0.0).

File: site/content/en/references/kustomize/kustomization/bases/_index.md (hunk 1 above)
Comment (natasha41575) [reply site/content/en/references/kustomize/kustomization/bases/_index.md:15]: Will update it!

Comment (natasha41575) [reply site/content/en/references/kustomize/kustomization/bases/_index.md]: Done.

File: site/content/en/references/kustomize/kustomization/patchesStrategicMerge/_index.md [hunk 2]
```diff
@@ -7,6 +7,19 @@ description: >
     Patch resources using the strategic merge patch standard.
 ---
 
+{{% pageinfo color="warning" %}}
+The `patchesStrategicMerge` field was deprecated in v5.0.0. The `bases` field was deprecated in v2.1.0. This field will never be removed from the
```
Comment (KnVerey) [inline site/content/en/references/kustomize/kustomization/patchesStrategicMerge/_index.md:11]:
```suggestion
The `patchesStrategicMerge` field was deprecated in v5.0.0. This field will never be removed from the
```

File: site/content/en/references/kustomize/kustomization/patchesjson6902/_index.md [hunk 3]
```diff
@@ -7,6 +7,19 @@ description: >
     Patch resources using the [json 6902 standard](https://tools.ietf.org/html/rfc6902)
 ---
 
+{{% pageinfo color="warning" %}}
+The `patchesJson6902` field was deprecated in v5.0.0. The `bases` field was deprecated in v2.1.0. This field will never be removed from the
```
Comment (KnVerey) [inline site/content/en/references/kustomize/kustomization/patchesjson6902/_index.md:11]:
```suggestion
The `patchesJson6902` field was deprecated in v5.0.0. This field will never be removed from the
```

File: site/content/en/references/kustomize/kustomization/vars/_index.md [hunk 4]
```diff
@@ -9,10 +9,19 @@ description: >
 
 [replacements]: https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/replacements/
 
-WARNING: There are plans to deprecate vars. For existing users of vars, we recommend migration to [replacements]
-as early as possible. There is a guide for convering vars to replacements at the bottom of this page under
-"convert vars to replacements". For new users, we recommend never using vars, and starting with replacements
-to avoid migration in the future.
+{{% pageinfo color="warning" %}}
+The `vars` field was deprecated in v5.0.0. The `bases` field was deprecated in v2.1.0. This field will never be removed from the
```
Comment (KnVerey) [inline site/content/en/references/kustomize/kustomization/vars/_index.md:13]:
```suggestion
The `vars` field was deprecated in v5.0.0.  This field will never be removed from the
```

**Review by KnVerey — COMMENTED** [review]
A couple copy-paste things to clean up. Should we mention `kustomize edit fix` for any of these?

File: site/content/en/references/kustomize/kustomization/patchesStrategicMerge/_index.md (hunk 2 above)
Comment (natasha41575) [reply site/content/en/references/kustomize/kustomization/patchesStrategicMerge/_index.md:11]: 🤦‍♀️ Sorry about that, I should be proof reading my own PRs...

**Comment by natasha41575** [comment]
> A couple copy-paste things to clean up. Should we mention kustomize edit fix for any of these?

Done! And added a warning that the automatic vars->replacement conversion may fail.

**Pull request merged**